enhancements/network-observability.md

```diff
@@ -0,0 +1,367 @@
+---
+title: network-observability
+authors:
+  - "@stleerh"
+  - "@jotak"
+  - "@mariomac"
+reviewers:
+  - "@russellb"
+  - "@mcurry-rh"
+  - "@bbennett"
+  - "@amorenoz"
+  - "@eraichst"
+  - "@eparis"
+  - "@spadgett"
+approvers:
+  - "@russellb"
+  - "@mcurry-rh"
+  - "@bbennett"
+creation-date: 2021-09-22
+last-updated: 2021-09-22
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Network Observability
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Network Observability introduces a new category in OpenShift that
+provides networking information for a single cluster.  It gives insight
+into what's on the network, when and what types of traffic and traffic
+flows are being made, and by whom.  It gathers data to help design, plan,
+and answer questions about the network and provides a visual representation
+to help understand, diagnose, and troubleshoot networking issues.
+
+## Motivation
+
+With Kubernetes, a layer of abstraction is added making it difficult for
+Red Hat and customers who managed their networks to be able to fully see
+what's happening on their network.  Monitoring provides metrics and
+alerts to potential problems.  Network observability will then help you
+analyze, investigate and diagnose those problems by looking at it from
+a control plane perspective instead of device by device.  In addition,
```

Comment:
I'd like to make a rather generic observation here:

At lest to me, Network Observability can mean two things (maybe it means both at the same time):
1 - Observing the network **traffic**
2 - Observing the network **logic**

What we're doing here is adding Traffic Network Observability since we're capturing information of packets that actually go through the network. But, in order to understand *why* that traffic went through (or why not), we would have to visualize the network's logic. That means observing 4 layers of network logic (from more to less abstract):
- k8s/ ovn-k8s level logic (probably covered by other parts of Openshift Console): What networks/pods/services/policies, etc are configured and how they relate to each other
- OVN logic: how the k8s logic gets translated into a set of Logical Routers, Logical Switches, Load Balancers, ACLs, etc and how these objects are connected to each other and to the k8s nodes (Chassis).
- OVS logic: the Openflow flows that OVN configures into OVS and define what it should do with packets as they come in and out of certain **openflow bridges  [1]**  (which are a logical entity that groups openflow flows together).
- Datapath logic: the datapath flows configured by OVS into the OVS kernel datapath. This is the logic that actually get's applied to flows that come through the system.

So, whenever someone wants to understand why a packet is being dealt in a specific way, they'll probably have to navigate some of these network logic layers.

Now, I think it's OK to focus this enhancement on network traffic observability but I'd say we should not forget about the rest. For instance, the process of determining what logic to "program" into the virtual network is commonly referred to as "control plane" so this sentence, in this particular context, seems a bit confusing to me.

**[1]** I'll refer to this logical bridge entity in another comment

Reply:
@stleerh we have [this](https://issues.redhat.com/browse/NETOBSERV-53) in JIRA, but indeed, this area could as well be mentioned in the "future work" section that Russell suggests